src/coreclr/jit/fginline.cpp

```diff
@@ -586,8 +606,58 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
                 }
 
                 CORINFO_CONTEXT_HANDLE contextInput = context;
+                context                             = nullptr;
                 m_compiler->impDevirtualizeCall(call, nullptr, &method, &methodFlags, &contextInput, &context,
                                                 isLateDevirtualization, explicitTailCall);
+
+                if (context != nullptr)
+                {
+                    Statement* newStmt = nullptr;
+                    GenTree**  callUse = nullptr;
+                    if (m_compiler->gtSplitTree(m_compiler->compCurBB, m_curStmt, call, &newStmt, &callUse))
```

Comment:
`gtSplitTree` needs to learn that it has to split `lvHasLdAddrOp` locals when it runs early.

Reply:
Done.